convert2rhel/unit_tests/subscription_test.py

```diff
@@ -306,3 +331,132 @@ def test_check_needed_repos_availability(self):
+
+        subscription.download_rhsm_pkgs()
+
+        self.assertEqual(subscription._download_rhsm_pkgs.called, 3)
+        self.assertEqual(subscription._download_rhsm_pkgs.pkgs_to_download,
+                         ["subscription-manager",
+                          "subscription-manager-rhsm-certificates",
+                          "python3-subscription-manager-rhsm",
+                          "dnf-plugin-subscription-manager",
+                          "python3-syspurpose"])
+
+    class StoreContentMocked(unit_tests.MockFunction):
+        def __init__(self):
+            self.called = 0
+            self.filename = None
+            self.content = None
+
+        def __call__(self, filename, content):
+            self.called += 1
+            self.filename = filename
+            self.content = content
+            return True
+
+    class DownloadPkgsMocked(unit_tests.MockFunction):
+        def __init__(self):
+            self.called = 0
+            self.to_return = ["/path/to.rpm"]
+
+        def __call__(self, pkgs, dest, reposdir=None):
+            self.called += 1
+            self.pkgs = pkgs
+            self.dest = dest
+            self.reposdir = reposdir
+            return self.to_return
+
+    @unit_tests.mock(system_info, "id", "centos")
+    @unit_tests.mock(system_info, "version", namedtuple("Version", ["major", "minor"])(7, 0))
+    @unit_tests.mock(utils, "store_content_to_file", StoreContentMocked())
+    @unit_tests.mock(utils, "download_pkgs", DownloadPkgsMocked())
+    def test__download_rhsm_pkgs(self):
+        subscription._download_rhsm_pkgs(["testpkg"], "/path/to.repo", "content")
+
+        self.assertEqual(utils.store_content_to_file.filename, None)
+        self.assertEqual(utils.download_pkgs.called, 1)
+        self.assertEqual(utils.download_pkgs.pkgs, ["testpkg"])
+
+        system_info.version = namedtuple("Version", ["major", "minor"])(6, 0)
+
+        subscription._download_rhsm_pkgs(["testpkg"], "/path/to.repo", "content")
+
+        self.assertTrue("/path/to.repo" in utils.store_content_to_file.filename)
+        self.assertEqual(utils.download_pkgs.called, 2)
+
+        utils.download_pkgs.to_return.append(None)
+
+        self.assertRaises(SystemExit, subscription._download_rhsm_pkgs, ["testpkg"], "/path/to.repo", "content")
+        
+
+    class DownloadPkgMocked(unit_tests.MockFunction):
+        def __init__(self):
+            self.called = 0
+            self.to_return = "/path/to.rpm"
+
+        def __call__(self, pkg, dest, reposdir=None):
+            self.called += 1
+            self.pkg = pkg
+            self.dest = dest
+            self.reposdir = reposdir
+            return self.to_return
+
+    @unit_tests.mock(utils, "download_pkg", DownloadPkgMocked())
+    @unit_tests.mock(utils, "run_subprocess", RunSubprocessMocked())
+    @unit_tests.mock(utils, "store_content_to_file", StoreContentMocked())
+    def test__get_rhsm_cert_on_centos_7(self):
```

Comment:
this test really tries to create the /etc/rhsm directory on host. This should be mocked